### PR TITLE
ACP-2128: Updated Vertex documentation

### PR DESCRIPTION
**Pull request by bohdan-turchyk-spryker**
## PR Description

- Updated Vertex documentation to follow 
- Please also include changes from https://github.com/spryker/spryker-docs/pull/2390

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md [hunk 1]
```diff
@@ -136,25 +140,67 @@ use Spryker\Zed\TaxApp\Communication\Plugin\Calculation\TaxAppCalculationPlugin;
 // ...
 ```
 
-{% info_block infoBox "Performance improvement" %}
+{% info_block infoBox "Updating from previous version" %}
 
-Spryker has its own [Taxes](/docs/pbc/all/tax-management/{{page.version}}/base-shop/tax-feature-overview.html) feature, which comes pre-installed in the Checkout through the Calculation module. To enhance performance when using an external Tax calculation provider, we recommend disabling the following plugins:
+Please pay attention to the position of TaxAppCalculationPlugin in the list of plugins.
```
Comment (matweew) [inline docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:145]: please add between what exactly plugins we expect to see 
(I understand that it's visible in the code snippet, but can clarify more)

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md [hunk 2]
```diff
@@ -136,25 +140,67 @@ use Spryker\Zed\TaxApp\Communication\Plugin\Calculation\TaxAppCalculationPlugin;
 // ...
 ```
 
-{% info_block infoBox "Performance improvement" %}
+{% info_block infoBox "Updating from previous version" %}
```
Comment (matweew) [inline docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:143]: specify what exactly version and version of what.

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md [hunk 3]
```diff
@@ -136,25 +140,67 @@ use Spryker\Zed\TaxApp\Communication\Plugin\Calculation\TaxAppCalculationPlugin;
...
+Please pay attention to the position of TaxAppCalculationPlugin in the list of plugins.
 
-in `\Pyz\Zed\Calculation\CalculationDependencyProvider::getQuoteCalculatorPluginStack()`:
+{% endinfo_block %}
 
-- TaxAmountCalculatorPlugin
-- ItemTaxAmountFullAggregatorPlugin
-- TaxRateAverageAggregatorPlugin
-- TaxTotalCalculatorPlugin
+Update `src/Pyz/Zed/TaxApp/TaxAppDependencyProvider.php` as follows:
```
Comment (matweew) [inline docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:149]: usually, if it's an old project it won't have this file, so customers need to **create** it. 
```suggestion
Create/update `src/Pyz/Zed/TaxApp/TaxAppDependencyProvider.php` as follows:
```

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md [hunk 4]
```diff
@@ -136,25 +140,67 @@ use Spryker\Zed\TaxApp\Communication\Plugin\Calculation\TaxAppCalculationPlugin;
...
+            new ItemTaxAmountFullAggregatorPlugin(),
+            new PriceToPayAggregatorPlugin(),
+            new TaxAmountAfterCancellationCalculatorPlugin(),
+        ];
+    }
+    
+// ...
+```
+
+In general, `getFallback<Quote|Order>CalculationPlugins` methods should contain the tax calculation plugins and all required plugins which are replaced by TaxAppCalculationPlugin in `\Pyz\Zed\Calculation\CalculationDependencyProvider`.
```
Comment (matweew) [inline docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:195]: this statement is not fully clear to me.
1. Please use `getFallbackQuoteCalculationPlugins()` and `getFallbackOrderCalculationPlugins()`.
2. `all required plugins which are replaced by TaxAppCalculationPlugin` The customers don't need to know the logic of this plugin, so we need to specify what plugins are needed or NOT needed and in what scenarios.

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md [hunk 5]
```diff
@@ -136,25 +140,67 @@ use Spryker\Zed\TaxApp\Communication\Plugin\Calculation\TaxAppCalculationPlugin;
...
+
+{% info_block infoBox "Fallback behavior" %}
+
+There are 3 different failure scenarios where TaxAppCalculationPlugin might need to use a fallback logic:
+
+1. TaxApp is not connected: fallback plugins defined in `getFallback<Quote|Order>CalculationPlugins` will be used.
+2. TaxApp is disabled: fallback plugins defined in `getFallback<Quote|Order>CalculationPlugins` will be used.
+3. TaxApp is not responding or is responding with an error: tax value will be set to zero and the customer will be able to proceed with the checkout.
```
Comment (matweew) [inline docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:203]: `TaxApp` -> `The Vertex app` (the document is about Vertex integration )

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md [hunk 6]
```diff
@@ -136,25 +140,67 @@ use Spryker\Zed\TaxApp\Communication\Plugin\Calculation\TaxAppCalculationPlugin;
...
 
-in `\Pyz\Zed\Calculation\CalculationDependencyProvider::getOrderCalculatorPluginStack()`:
+```php
+// ...
 
-- TaxAmountCalculatorPlugin
-- ItemTaxAmountFullAggregatorPlugin
-- TaxAmountAfterCancellationCalculatorPlugin
-- OrderTaxTotalCalculationPlugin
+use ;
```
Comment (matweew) [inline docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:154]: Please add full content in the code snippet including all `use` statements.

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md [hunk 7]
```diff
@@ -235,7 +281,35 @@ class MessageBrokerDependencyProvider extends SprykerMessageBrokerDependencyProv
...
+{
+// ...
+
+    /**
+     * @return array<\Spryker\Zed\OmsExtension\Dependency\Plugin\OmsEventTriggeredListenerPlugin>
+     */
+    protected function getOmsEventTriggeredListenerPlugins(): array
+    {
+        return [
+            new OrderRefundedEventListenerPlugin(),
```
Comment (matweew) [inline docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:305]: Please add a note, that this plugin expects a specific state name in the OMS configuration to work.

Maybe can be merged with #6 to the section `Configure your OMS process.`

**Review by matweew — CHANGES_REQUESTED** [review]

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md [hunk 8]
```diff
@@ -235,7 +281,35 @@ class MessageBrokerDependencyProvider extends SprykerMessageBrokerDependencyProv
 
 ```
 
-### 5. Optional: To send invoices to Vertex through OMS, configure your Payment OMS
+### 5. Configure the OMS dependency provider
+
+Add the following code to `src/Pyz/Zed/Oms/OmsDependencyProvider.php`:
```
Comment (matweew) [inline docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:286]: BTW in the next section we already have these words

>Configure the Oms dependency provider
>Add the config to src/Pyz/Zed/Oms/OmsDependencyProvider.php:

**Comment by matweew** [comment]
@bohdan-turchyk-spryker Please update `last_updated:` on updated pages

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md (hunk 4 above)
Comment (bohdan-turchyk-spryker) [reply docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:195]: as for 2: the idea here is that we ask the customer to move all the tax-related plugins (except for TaxTotal) into the "fallback" stacks
we can list the OOTB tax plugins we expect them to move but there might be other plugins they have there, right?

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md (hunk 7 above)
Comment (bohdan-turchyk-spryker) [reply docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:305]: Section 6 is marked as optional as it covers invoices. I assume that we aren't marketing refunds as optional feature?

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md (hunk 8 above)
Comment (bohdan-turchyk-spryker) [reply docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:286]: This section is marked as optional and I'm not sure we treat refunds as optional feature.

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md (hunk 7 above)
Comment (matweew) [reply docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:310]: Please check it with Oluwanifemi Omotola

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md (hunk 4 above)
Comment (matweew) [reply docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md]: they might have, but the change is pretty low, so I would list all needed OOTB Spryker plugins (BTW please change OOTB to full name :) ). Let's don't mention custom plugins there at all, I expect that if project decides to create calculation plugins they understand how to deal with them and debug them.

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md (hunk 7 above)
Comment (bohdan-turchyk-spryker) [reply docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:310]: Refunds are optional, moved the config section

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md (hunk 8 above)
Comment (bohdan-turchyk-spryker) [reply docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:291]: Refunds are optional, moved the config section

**Review by matweew — DISMISSED** [review]

**Review by matweew — DISMISSED** [review]

File: docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md [hunk 9]
```diff
@@ -28,674 +28,10 @@ Ensure that your installation meets these requirements.
...
-| city             | city           |                                    |
-| state            | mainDivision   | Should be either not empty or null |
-| zipCode          | postalCode     |                                    |
-| country.iso2Code | country        |                                    |
+1. [Integrate ACP connector module for tax calculation](/docs/pbc/all/tax-management/{{page.version}}/base-shop/third-party-integrations/vertex/integrate-connector-module.html)
+2. [Integrate Vertex app](/docs/pbc/all/tax-management/{{page.version}}/base-shop/third-party-integrations/vertex/integrate-vertex-app.html)
 
 ## Next step
-[Configure Vertex in the Back Office](/docs/pbc/all/tax-management/{{page.version}}/base-shop/third-party-integrations/vertex/configure-vertex.html)
+[Configure Vertex in the Back Office](/docs/pbc/all/tax-management/{{page.version}}/base-shop/third-party-integrations/vertex/configure-vertex.html)
```
Comment (bohdan-turchyk-spryker) [inline docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md:37]: I've kept this "next step" section here as it is a next step after installation is finished; at the same time the same section is present in "integrate-vertex-app.md" file. Should we keep it like this or should it be present in both files?

Also: should we keep the folder structure like this or should new files be moved lower in the hierarchy?

**Pull request merged**